src/Workspaces/MSBuildTest/OpenProjectsTests.cs

```diff
@@ -0,0 +1,222 @@
+
+            var compilation = await project.GetCompilationAsync();
+            Assert.NotNull(compilation);
+
+            // Unnecessary using directives are reported with a severity of Hidden
+            var diagnostics = compilation!.GetDiagnostics()
+                .Where(diagnostic => diagnostic.Severity > DiagnosticSeverity.Hidden && ignoredDiagnostics.Contains(diagnostic.Id) != true);
+
+            Assert.Empty(diagnostics);
+        }
+
+        private void CleanupProject(string templateName, string languageName)
+        {
+            var projectPath = GetProjectPath(templateName, languageName);
+
+            if (Directory.Exists(projectPath))
+            {
+                Directory.Delete(projectPath, true);
+            }
+        }
+
+        private string GetProjectPath(string templateName, string languageName)
+        {
+            var languagePrefix = languageName.Replace("#", "Sharp").Replace(' ', '_').ToLower();
+            var projectName = $"{languagePrefix}_{templateName}_project";
+            return Path.Combine(SolutionDirectory.Path, projectName);
+        }
+
+        private static string GetProjectFilePath(string projectPath, string languageName)
+        {
+            var projectName = Path.GetFileName(projectPath);
+            var projectExtension = languageName switch
+            {
+                LanguageNames.CSharp => "csproj",
+                LanguageNames.VisualBasic => "vbproj",
+                _ => throw new ArgumentOutOfRangeException(nameof(languageName), actualValue: languageName, message: "Only C# and VB.Net project are supported.")
+            };
+            return Path.Combine(projectPath, $"{projectName}.{projectExtension}");
+        }
+
+        public int NewProject(string templateName, string outputPath, string languageName, ITestOutputHelper output)
+        {
+            var language = languageName switch
+            {
+                LanguageNames.CSharp => "C#",
+                LanguageNames.VisualBasic => "VB",
+                LanguageNames.FSharp => "F#",
+                _ => throw new ArgumentOutOfRangeException(nameof(languageName), actualValue: languageName, message: "Only C#, F# and VB.NET project are supported.")
+            };
+
+            var restoreResult = RunDotNet($"new \"{templateName}\" -o \"{outputPath}\" --language \"{language}\"");
+
+            output.WriteLine(string.Join(Environment.NewLine, restoreResult.Output));
+
+            return restoreResult.ExitCode;
+        }
+
+        private ProcessResult RunDotNet(string arguments)
+        {
+            Assert.NotNull(DotNetSdkMSBuildInstalled.SdkPath);
+
+            var dotNetExeName = "dotnet" + (Path.DirectorySeparatorChar == '/' ? "" : ".exe");
+            var exePath = Path.Combine(DotNetSdkMSBuildInstalled.SdkPath!, dotNetExeName);
+
+            return ProcessUtilities.Run(
+                exePath, arguments,
+                workingDirectory: SolutionDirectory.Path);
```

Comment:
What is SolutionDirectory.Path in this case? Is that different than the output path that's being passed from NewProject? (And if it's different, why not just create an empty folder and have that get invoked directly?)

Reply:
SolutionDirectory is a temporary directory inherited from the WorkspaceTestBase. It is the parent directory of the project directories that we create. Since, the projectPath argument to NewProject is a full path, it is likely unnecessary to set the workingDirectory.

Reply:
Removed